service/src/main/java/org/databiosphere/workspacedataservice/SentryInitializer.java

```diff
@@ -40,26 +37,21 @@ public class SentryInitializer {
   @Value("${sentry.deploymentMode}")
   String deploymentMode;
 
-  private static final Pattern SAM_ENV_PATTERN = Pattern.compile("\\.dsde-(\\p{Alnum}+)\\.");
   private static final String DEFAULT_ENV = "unknown";
+
   // Environments we want to monitor on sentry - don't send errors from local, bees, or Github
   // actions
-  private static final List<String> environments = List.of("prod", "staging", "dev");
+  private static final List<String> sentryEnvironments = List.of("prod", "staging", "dev");
 
   @Bean
-  public SmartInitializingSingleton initialize() {
-    String env;
-    if (StringUtils.isNotBlank(terraEnv)) {
-      env = terraEnv;
-    } else {
-      env = urlToEnv(samurl);
-    }
+  public SmartInitializingSingleton initialize(Environment environment) {
+    var env = getSentryEnvironment(environment.getActiveProfiles()).orElse(DEFAULT_ENV);
 
     return () ->
         Sentry.init(
             options -> {
               options.setEnvironment(env);
-              options.setDsn(environments.contains(env) ? dsn : "");
+              options.setDsn(!env.equals(DEFAULT_ENV) ? dsn : "");
```

Comment:
Nit: this could be simplified by swapping the branches.

```suggestion
              options.setDsn(env.equals(DEFAULT_ENV) ? "" : dsn);
```